### PR TITLE
Update product detail page

### PR DESCRIPTION
**Pull request by priyankaarkade**
- Updated detail page when the product is null use loading image or show product detail.
- When the user clicks the product item, it will navigates to individual product detail page.
- Added some css to look good.
- Resolved issue regarding formatting.
- Let me know if i need to change or refactor code.

### CODE REVIEW

File: .idea/encodings.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="Encoding" addBOMForNewFiles="with NO BOM" />
+</project>
```
Comment (kyle-arkade) [inline .idea/encodings.xml]: You should add `.idea` folder to your `.gitignore` file.

File: src/containers/ProductDetail/index.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+class ProductDetail extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            product: null
+        };
+    }
+
+    componentDidMount() {
+        let Id = this.props.match.params.id
```
Comment (kyle-arkade) [inline src/containers/ProductDetail/index.js]: There is no need to assign ID to a variable here. Just do `this.getProduct(this.props.match.params.id);`.

File: src/containers/ProductDetail/index.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                    <h2>{this.state.product.description}</h2>
+                </div>
+                <h3 className="product__price">${this.state.product.variants[0].price}</h3>
+            </div>
+            )
+        }
+    }
+}
+
+export default withRouter(ProductDetail);
```
Comment (kyle-arkade) [inline src/containers/ProductDetail/index.js]: Put a blank line at the end of each file.

File: src/containers/ProductsCard/index.js [hunk 4]
```diff
@@ -17,33 +17,32 @@ class ProductsCard extends Component {
 
     getProducts() {
         Sdk.product.fetchAll().then((products) => {
-            console.log("allproducts",products);
             this.setState({
                 products: products
             })
-            console.log('product updated state:', this.state.products);
+            console.log('allproduct updated state:', this.state.products);
```
Comment (kyle-arkade) [inline src/containers/ProductsCard/index.js]: Normally remove console logs before merging into master.

File: src/index.js [hunk 5]
```diff
@@ -1,35 +1,28 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import App from './App';
-import Client from 'shopify-buy';
 import './index.css';
-// import * as serviceWorker from './serviceWorker';
-// import ProductCard from './containers/ProductCard/index';
-// import ProductsCard from './containers/ProductsCard/index';
-// import { Route, Redirect, BrowserRouter as Router } from 'react-router-dom';
+import * as serviceWorker from './serviceWorker';
```
Comment (kyle-arkade) [inline src/index.js]: Not sure what service worker is for

File: src/containers/ProductDetail/index.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    getProduct = (productId) => {
+        Sdk.product.fetch(productId).then((product) => {
+            this.setState({
+                product: product
+            })
+        }) 
+    }
+
+    render() {
+
```
Comment (kyle-arkade) [inline src/containers/ProductDetail/index.js]: No blank line

File: src/containers/ProductDetail/index.js (hunk 3 above)
Comment (priyankaarkade) [reply src/containers/ProductDetail/index.js]: added blank line at the end of each file.

File: src/containers/ProductDetail/index.js (hunk 2 above)
Comment (priyankaarkade) [reply src/containers/ProductDetail/index.js]: Removed variable and added as `this.getProduct(this.props.match.params.id);`

File: .idea/encodings.xml (hunk 1 above)
Comment (priyankaarkade) [reply .idea/encodings.xml]: added .idea in .gitignore file.

File: src/containers/ProductsCard/index.js (hunk 4 above)
Comment (priyankaarkade) [reply src/containers/ProductsCard/index.js]: removed comments.

File: src/index.js (hunk 5 above)
Comment (priyankaarkade) [reply src/index.js]: Removed service worker as it is unused.

File: src/containers/ProductDetail/index.js (hunk 6 above)
Comment (priyankaarkade) [reply src/containers/ProductDetail/index.js]: deleted blank line.

File: src/containers/ProductsCard/index.js [hunk 7]
```diff
@@ -17,37 +17,35 @@ class ProductsCard extends Component {
...
-        const product = this.state.products.map((product,index) => {
+        const product = this.state.products.map((product, index) => {
             return(
-                <div key={index} className="products-card">
-                    <div className="products-image">
-                        <img src={product.images[0].src} alt="images" width="200" height="200"/>
-                    </div>
-                    <div className="products-info">
-                        <Link to={'/products/' + product.id}>
+                <Link  className="product__link" to={`/products/${product.id}`} key={product.id}>
```
Comment (kyle-arkade) [inline src/containers/ProductsCard/index.js]: There       is       an     extra <Space    here.

**Pull request merged**